src/Microsoft.Tye.Core/ApplicationFactory.cs

```diff
@@ -21,227 +21,257 @@ public static async Task<ApplicationBuilder> CreateAsync(OutputContext output, F
+                            UseMultiphaseDockerfile = false,
+                        };
+
+                        // Do k8s by default.
+                        project.ManifestInfo = new KubernetesManifestInfo();
+                    }
+                    else if (!string.IsNullOrEmpty(configService.Image))
                     {
-                        project.EnvironmentVariables.Add(envVar);
+                        var container = new ContainerServiceBuilder(configService.Name!, configService.Image)
+                        {
+                            Args = configService.Args,
+                            Replicas = configService.Replicas ?? 1
+                        };
+                        service = container;
                     }
-                    else if (service is ContainerServiceBuilder container)
+                    else if (!string.IsNullOrEmpty(configService.Executable))
                     {
-                        container.EnvironmentVariables.Add(envVar);
+                        var expandedExecutable = Environment.ExpandEnvironmentVariables(configService.Executable);
+                        var workingDirectory = "";
+
+                        // Special handling of .dlls as executables (it will be executed as dotnet {dll})
+                        if (Path.GetExtension(expandedExecutable) == ".dll")
+                        {
+                            expandedExecutable = Path.GetFullPath(Path.Combine(config.Source.Directory.FullName, expandedExecutable));
+                            workingDirectory = Path.GetDirectoryName(expandedExecutable)!;
+                        }
+
+                        var executable = new ExecutableServiceBuilder(configService.Name!, expandedExecutable)
+                        {
+                            Args = configService.Args,
+                            WorkingDirectory = configService.WorkingDirectory != null ?
+                            Path.GetFullPath(Path.Combine(config.Source.Directory.FullName, Environment.ExpandEnvironmentVariables(configService.WorkingDirectory))) :
+                            workingDirectory,
+                            Replicas = configService.Replicas ?? 1
+                        };
+                        service = executable;
                     }
-                    else if (service is ExecutableServiceBuilder executable)
+                    else if (!string.IsNullOrEmpty(configService.Import))
                     {
-                        executable.EnvironmentVariables.Add(envVar);
+                        var expandedYaml = Environment.ExpandEnvironmentVariables(configService.Import);
+                        var nestedConfig = ConfigFactory.FromFile(new FileInfo(Path.Combine(config.Source.DirectoryName, expandedYaml)));
+                        nestedConfig.Validate();
+                        queue.Enqueue(nestedConfig);
+                        // TODO do we need a service builder here?
```

Comment:
this should be represented by a service builder, but I would expect it to be one of the existing types.

Reply:
If there is a reference to a specific service in a tye.yaml, for example:

```yaml
name: VotingSample
services:
- name: include
  include: ../worker/tye.yaml
- name: results
  project: results.csproj
```

```yaml
name: VotingSample
services:
- name: redis
  image: redis
  bindings:
    - port: 6379
      connectionString: ${host}:${port}
- name: worker
  project: worker.csproj
- name: postgres
  image:  postgres
  env:
  - name: POSTGRES_PASSWORD
    value: "pass@word1"
  bindings:
  - port: 5432
    connectionString: Server=${host};Port=${port};User Id=postgres;Password=${env:POSTGRES_PASSWORD};

```

We would have two service builders with the same name (worker) because we added a service builder for worker in both the results tye.yaml and the worker tye.yaml. Hence I didn't do it here yet, but wanted feedback/thoughts.